src/cmd/ship.go

```diff
@@ -135,6 +136,10 @@ func getShipStepList(config shipConfig) (steps.StepList, error) {
 		result.Append(&steps.PushBranchStep{BranchName: branchToMergeInto, Undoable: true})
 	}
 	childBranches := git.Config().GetChildBranches(config.BranchToShip)
+	// NOTE: when shipping with a driver, we can always delete the remote branch because:
+	// - we know we have a tracking branch (otherwise there would be no PR to ship via driver)
+	// - we have updated the PRs of all child branches (because we have API access)
+	// - we know we are online
```

Comment:
Feel conflicted about whether this is needed. The code reads that we can delete the remote branch if `canShipWithDriver` is true or we have a tracking branch, have no children, and are online. 

If we are going to have this comment I'd rather move it to the `getCanShipWithDriver` that explains what the implications of it are (how we merge and that we can always delete the remote branch)